File: test-src/27_map.kt
import java.util.*

fun main(args: Array<String>) {
  // Constructors
  val hashMap0 = HashMap<String, String>()
  val linkedHashMap0 = LinkedHashMap<String, String>()
  val map0 = emptyMap<String, String>()
  val map1 = mapOf(Pair(10, "hi"))
  val map2 = mapOf(Pair("as", "hi"), Pair("df", "hello"), Pair("gh", "salut"))
  var hashMap = hashMapOf(Pair(1, "hi"))
  var linkedMap = linkedMapOf(Pair(1, "hi"))
  var mutableMap = mutableMapOf(Pair(1, "hi"))

  // Basic calls
  println("${map1[10]} (hi)")
  println("${map2["as"]} (hi)")
  println("${map2.size} (3)")
  hashMap.put(2, "hello")
  linkedMap.putAll(linkedMap)
  mutableMap.remove(1)

  println(hashMap)
  if (hashMap.size != 2) {
    println("ERROR: hashMap.size")
  }
  println(linkedMap)
  if (linkedMap.size != 1) {
    println("ERROR: linkedMap.size")
  }
  println(mutableMap)
  if (mutableMap.size != 0) {
    println("ERROR: mutableMap.size")
  }

  if (!map0.isEmpty()) {
    println("ERROR: map0.isEmpty()")
  }

  println("${hashMap.keys} ([2, 1])")
  println("${hashMap.values} ([hello, hi])")

  // Iteration
  for ((k, v) in map2) {
    println("$k : $v")
  }
}

File: test-dest/27_map.swift

func main(args: [String]) {
  // Constructors
  var hashMap0 = [String: String]()
  var linkedHashMap0 = [String: String]()
  var map0 = [String: String]()
  var map1 = [10: "hi"]
  var map2 = ["as": "hi", "df": "hello", "gh": "salut"]
  var hashMap = [1: "hi"]
  var linkedMap = [1: "hi"]
  var mutableMap = [1: "hi"]

  // Basic calls
  print("\(map1[10]) (hi)")
  print("\(map2["as"]) (hi)")
  print("\(map2.size) (3)")
  hashMap.put(2, "hello")
  linkedMap.putAll(linkedMap)
  mutableMap.remove(1)

  print(hashMap)
  if hashMap.size != 2 {
    print("ERROR: hashMap.size")
  }
  print(linkedMap)
  if linkedMap.size != 1 {
    print("ERROR: linkedMap.size")
  }
  print(mutableMap)
  if mutableMap.size != 0 {
    print("ERROR: mutableMap.size")
  }

  if !map0.isEmpty() {
    print("ERROR: map0.isEmpty()")
  }

  print("\(hashMap.keys) ([2, 1])")
  print("\(hashMap.values) ([hello, hi])")

  // Iteration
  for (k, v) in map2 {
    print("\(k) : \(v)")
  }
}

main([])
